analysis_options.yaml

```diff
@@ -1,16 +1,5 @@
 include: package:lints/recommended.yaml
 
-analyzer:
-  exclude:
-    - "**/*.g.dart"
-  errors:
-    omit_local_variable_types: ignore
-    annotate_overrides: ignore
-    use_function_type_syntax_for_parameters: ignore
-    missing_required_param: warning
-    missing_return: warning
-    prefer_single_quotes: warning
```

Comment:
I would definitely leave these 3 inside

Reply:
I assumed that not setting parameters meant leaving the default values: is it not the case?
The point of this PR is to match the pub.dev settings, so the less we set ourselves, the better.

Reply:
Probably is, nevertheless, i am in favor of stronger rules. These will probably not deduct points

Reply:
How do you know "our" rules are stronger and will always be stronger? They change the rules regularly. Maybe one day they will say "missing_return: error" instead of "warning", and we would miss it if we override with only "warning".

If I started this PR it's because our rules were less strict (more specifically, we ignored generated files, they don't).

In addition to that - but it's not 100% related to this PR - I think there are several rules we apply in Smoothie that are neither useful nor relevant, e.g. having to specify the type of an empty `List`.
Like `List<int> list = <int>[];`
I prefer `List<int> list = [];`, easier to write and to read, and I don't see what we're missing here. But that's another story.